src/renderers/index.js

```diff
@@ -0,0 +1,174 @@
+      throw new Error(`Renderer name must be a string, received ${name}`)
+    }
+    if (name.startsWith('#')) {
+      name = name.substr(1)
+    }
+    return !!renderers[name]
+  }
+
+  const invoke = ({ rendererFn, rendererName, context, outputPlatform, incomingEvent = null }) => {
+    // TODO throw if incomingEvents null <<<==== MOCK IT
+
+    const options = {
+      throwIfNoPlatform: true,
+      currentPlatform: outputPlatform
+    }
+
+    return Engine({ rendererFn, rendererName, context, options, processors, incomingEvent })
+  }
+
+  const doSendContent = (rendererFn, { rendererName, context, outputPlatform, incomingEvent }) => {
+    const messages = invoke({ rendererFn, rendererName, context, outputPlatform, incomingEvent })
+
+    return Promise.mapSeries(messages, message => {
+      if (message.__internal) {
+        if (message.type === 'wait') {
+          return Promise.delay(message.wait)
+        }
+      } else {
+        return middlewares.sendOutgoing(message)
+      }
+    })
+  }
+
+  const sendContent = async (incomingEvent, rendererName, additionalData = {}) => {
+    rendererName = rendererName.startsWith('#') ? rendererName.substr(1) : rendererName
+
+    const initialData = {}
+
+    if (rendererName.startsWith('!')) {
+      const itemName = rendererName.substr(1)
+      const contentItem = await contentManager.getItem(itemName)
+
+      if (!contentItem) {
+        throw new Error(`Could not find content item with ID "${itemName}" in the Content Manager`)
+      }
+
+      const { categoryId: itemCategoryId } = contentItem
+
+      const itemCategory = contentManager.getCategorySchema(itemCategoryId)
+
+      if (!itemCategory) {
+        throw new Error(
+          `Could not find category "${itemCategoryId}" in the Content Manager` + ` for item with ID "${itemName}"`
+        )
+      }
+
+      const itemRenderer = itemCategory.renderer
+      if (!_.isString(itemRenderer) || !itemRenderer.startsWith('#') || itemRenderer.length <= 1) {
+        throw new Error(`Invalid renderer '${itemRenderer}' in category '${itemCategoryId}' of Content Manager.
+         A renderer must start with '#'`)
+      }
+
+      rendererName = itemRenderer.substr(1)
+      Object.assign(initialData, contentItem.data)
+    }
+
+    // TODO Add more context
+    const fullContext = Object.assign(
+      {},
+      initialData,
+      {
+        user: incomingEvent.user,
+        originalEvent: incomingEvent
+      },
+      additionalData
+    )
+
+    const renderer = renderers[rendererName]
+
+    if (!renderer) {
+      const error = `[Renderer] Renderer not defined (#${rendererName})`
+      logger.error(error)
+      throw new Error(error)
+    }
+
+    return doSendContent(renderer, {
+      rendererName,
+      context: fullContext,
+      outputPlatform: incomingEvent.platform,
+      incomingEvent
+    })
+  }
+
+  const processIncoming = (event, next) => {
+    event.reply = (blocName, additionalData = {}) => {
```

Comment:
`blocName` -> `rendererName`.
Also there are things like `rawBloc` in the previous file